### DISCUSSION TITLE
Firebase notifications not received in lock screen

### DISCUSSION THREAD

**Discussion by VeerammalV** [community/community#140985] [Programming Help] [answered]
### Body

I'm using firebase for receiving push notifications (notification payload). The notifications are received when the app is in background, foreground or even closed. When my mobile screen is locked, the notifications are received for about 10-15 minutes. After that I don't receive any messages at all. But when I unlock my phone, all the notifications come at once. I don't know why notifications are not showing in lock screen after some times.. I want to receive notifications at any times even when the screen is locked.. I've tried this with Power manager, Battery optimisation, Direct boot aware etc. none of them worked. If anyone know the answer, plz comment. Thanks in advance.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by tpmst** [comment]
Hello @VeerammalV 

have you tried sending the message with high priority? If not here is a quick explanation:

**Comment by tpmst** [comment]
@VeerammalV sorry i hit shift and enter :)

### High Priority FCM Messages:
Make sure your Firebase Cloud Messaging (FCM) notifications are sent with high priority. This can be done by setting the priority field of your notification to "high" when sending messages from your server. For example:
```
{
  "to": "<FCM_TOKEN>",
  "priority": "high",
  "notification": {
    "title": "Hello",
    "body": "World"
  }
}
```
High-priority messages can wake the device from Doze mode to process the incoming notification.

**Reply by VeerammalV** [reply]
No, It didn't work after setting high priority..

**Reply by tpmst** [reply, marked as answer]
Hello @VeerammalV

maby it could be the Battery Optimization Settings. In Android settings, you can select your app under `Battery Optimization` and choose `Don't optimize`. This tells the system not to restrict your app’s background activities, which can help with receiving notifications on time.

If your app needs to perform background work at regular intervals and ensure timely delivery of notifications, consider using AlarmManager with `setExactAndAllowWhileIdle` or `setAndAllowWhileIdle` methods to schedule alarms that can operate even during Doze mode.


Or maby if applicable and without draining the battery excessively, you might consider sending keep-alive messages to ensure the network connection is maintained. This is more complex and should be used sparingly as it can lead to increased battery consumption.

**Reply by VeerammalV** [reply]
Thank you..😊
